src/shared/Atlassian.Bitbucket/BitbucketAuthentication.cs

```diff
@@ -211,29 +216,28 @@ public async Task<bool> ShowOAuthRequiredPromptAsync()
             }
         }
 
-        public async Task<OAuth2TokenResult> CreateOAuthCredentialsAsync(Uri targetUri)
+        public async Task<OAuth2TokenResult> CreateOAuthCredentialsAsync(InputArguments input)
         {
-            ThrowIfUserInteractionDisabled();
-
-            var oauthClient = new BitbucketOAuth2Client(HttpClient, Context.Settings);
-
             var browserOptions = new OAuth2WebBrowserOptions
             {
                 SuccessResponseHtml = BitbucketResources.AuthenticationResponseSuccessHtml,
                 FailureResponseHtmlFormat = BitbucketResources.AuthenticationResponseFailureHtmlFormat
             };
 
             var browser = new OAuth2SystemWebBrowser(Context.Environment, browserOptions);
-            var authCodeResult = await oauthClient.GetAuthorizationCodeAsync(Scopes, browser, CancellationToken.None);
+            var authCodeResult = await oauth2ClientRegistry.Get(input).GetAuthorizationCodeAsync(oauth2ClientRegistry.Get(input).Scopes, browser, CancellationToken.None);
```

Comment:
Please break out some of these inline method calls for readability :)

```csharp
var client = oauth2ClientRegistry.Get(input);
var authCodeResult = await client.GetAuthorizationCodeAsync(client.Scopes, browser, CancellationToken.None);
```

Also see here how this looks weird to pass `client.Scopes` back to `client`?